discovery/sync_manager.go

```diff
@@ -262,14 +262,13 @@ func (m *SyncManager) syncerHandler() {
 			// attemptHistoricalSync determines whether we should
 			// attempt an initial historical sync when a new peer
 			// connects.
-			attemptHistoricalSync := false
+			attemptHistoricalSync := true
```

Comment:
This doesn't really match the PR title/body: this'll have us _always_ do a historical sync for _all_ newly connected peers, which might also include flappy (connecting and disconnecting in a tight time frame) peers.

Reply:
Also if we already have enough active syncers (line 290 below), then this won't have any effect.

Reply:
yeah sorry I haven't figured out a proper fix so don't review it yet. Was working on something else then realized this issue so open a PR but realized it wasn't the fix so turned it into a draft😂